pycqed/instrument_drivers/physical_instruments/QuTech_VSM_Module.py

```diff
@@ -0,0 +1,267 @@
+                               set_cmd='MARKER:' + mod_scpi + ':SOURCE {}',
+                               vals=vals.Enum('INT', 'EXT'))
+            self.add_parameter(mod_name + '_marker_state',
+                               set_cmd='MARKER:'+mod_scpi+':STATE {}',
+                               vals=vals.OnOff())
+
+        for channel in self.channels:
+            for mod in self.modules:
+                mod_ch_name = 'mod{m}_ch{c}'.format(m=mod, c=channel)
+                mod_ch_scpi = 'MODULE{m}:CHANNEL{c}'.format(m=mod, c=channel)
+                self.add_parameter(mod_ch_name + '_marker_state',
+                                   get_cmd='MARKER:'+mod_ch_scpi+':STATE?',
+                                   set_cmd='MARKER:'+mod_ch_scpi+':STATE {}',
+                                   vals=vals.OnOff())
+                self.add_parameter(mod_ch_name + '_marker_source',
+                                   get_cmd='MARKER:'+mod_ch_scpi+':SOURCE?',
+                                   vals=vals.Enum('int', 'ext'))
+
+    def add_qubit_parameters(self):
+        # Command class qubit
+        for channel in self.channels:
+            ch_name = '_ch{c}'.format(c=channel)
+            ch_scpi = ':CHANNEL{c}'.format(c=channel)
+            self.add_parameter('qubit' + ch_name + '_description',
+                               get_cmd='QUBIT'+ch_scpi+':DESCRIPTION?',
+                               set_cmd='QUBIT'+ch_scpi+':DESCRIPTION {}',
+                               vals=vals.Strings())
+            self.add_parameter('qubit' + ch_name + '_frequency',
+                               get_cmd='QUBIT'+ch_scpi+':FREQUENCY?',
+                               set_cmd='QUBIT'+ch_scpi+':FREQUENCY {}',
+                               vals=vals.Numbers())
+            self.add_parameter('qubit' + ch_name + '_led_color',
+                               get_cmd='QUBIT'+ch_scpi+':LEDCOLOR?',
+                               set_cmd='QUBIT'+ch_scpi+':LEDCOLOR {}',
+                               vals=vals.Strings())
+            for mod in self.modules:
+                mod_ch_name = '_mod{m}_ch{c}'.format(m=mod, c=channel)
+                mod_ch_scpi = ':MODULE{m}:CHANNEL{c}'.format(m=mod, c=channel)
+                self.add_parameter('qubit' + mod_ch_name,
+                                   get_cmd='QUBIT'+mod_ch_scpi+'?',
+                                   set_cmd='QUBIT'+mod_ch_scpi+' {}',
+                                   vals=vals.OnOff())
+
+    def add_parameters(self):
+        # VSM properties
+
+        self.add_power_supply_parameters()
+        self.add_parameter('mbbc_state', get_cmd='MBBC?',
+                           vals=vals.Enum('connected', 'disconnected'))
+        self.add_temperature_parameters()
+        self.add_marker_parameters()
+        self.add_qubit_parameters()
+
+        # Command class calibration
+        for pulse in ('GAUSSIAN', 'DERIVATIVE'):
+            for ap in ('ATTENUATION', 'PHASE'):
```

Comment:
I can see the calibration paramter but I can find nowhere where we can set the channel specific attenuation/phases. Also (once these are added) what units are these in? dB? raw dac values?